CMP1903M Assessment 1 Base Code/Program.cs

```diff
@@ -18,42 +18,93 @@ class Program
+                    string? manual;
+                    while (true)
+                    {
+                        manual = Console.ReadLine();
+
+                        if (manual == null)
+                        {
+                            Console.WriteLine("The input is invalid, please try again: ");
+                        }
+                        else
+                        {
+                            var success = input.ManualTextInput(manual);
+
+                            if (success)
+                            {
+                                break;
+                            } else
+                            {
+                                Console.WriteLine("Please try again: ");
+                            }
+                        }
+                    }
 
+                    analysis.Analysis(input);
+
+                    break;
                 }
-                
-                else if (input == "2")
+
+                else if (input_mode == "2")
                 {
-                    string Text = System.IO.File.ReadAllText("Text.txt"); //You will have to change this to where you place a text file
-                    Input.FileTextInput(Text);
+                    Console.WriteLine("\nPlease enter the path to file input: ");
+
+                    // ask user for a valid path
+                    string? file_path;
+                    while (true)
+                    {
+                        file_path = Console.ReadLine();
 
+                        // different errors that could happen
+                        if (file_path == null)
+                        {
+                            Console.WriteLine("Input is invalid, please try again: ");
+                        }
+                        else if (!File.Exists(file_path))
+                        {
+                            Console.WriteLine("File doesn't exist, please input a valid file: ");
+                        }
+                        else
+                        {
+                            var input_success = input.FileTextInput(file_path);
+                            if (input_success) {
+                                break;
+                            } else
+                            {
+                                Console.WriteLine("Please try again: ");
+                            }
+                        }
+                    }
+
+                    input.FileTextInput(file_path);
+
+                    analysis.Analysis(input);
+
+                    break;
                 }
-                else if (input != "2" || input != "1") //If input is not 1 or 2, Program resets
+                else if (input_mode != "2" || input_mode != "1") //If input is not 1 or 2, Program resets
                 {
-                    Console.WriteLine("\nPlease enter 1 or 2");
-                    Program.Main();
+                    Console.WriteLine("Invalid input, please try again: ");
```

Comment:
As mentioned before, I recommend removing the `while(true)` loop at the top of the file and instead placing a call to `Main()` below:
```suggestion
                    Console.WriteLine("Invalid input, please try again: ");
                    Main();
```